pkgs/os-specific/darwin/apple-sdk/mk-stub.nix

```diff
@@ -1,14 +1,26 @@
-{ stdenvNoCC }:
+{ lib, stdenvNoCC }:
 
-version: pname:
-stdenvNoCC.mkDerivation {
-  inherit pname version;
+let
+  common = "these stubs do nothing and will be removed in Nixpkgs 25.11; see <https://nixos.org/manual/nixpkgs/stable/#sec-darwin> for documentation and migration instructions.";
+in
+msg: version: pname:
+lib.warnOnInstantiate
+  (
+    if builtins.isString msg then
+      "${msg}.${pname}: ${common}"
+    else
+      "${msg.prefix}.${pname}: ${common} ${msg.suffix}"
```

Comment:
This is just for “Add `apple-sdk_12` to build inputs instead to use the macOS 12 SDK.”, right? I think we can skip the complexity here: the migration instructions tell you how to use the new SDK pattern, and few people will be using the macOS 12 SDK with the old pattern to begin with anyway. (But you did already implement it, so it’s fine if you don’t want to touch it again.)

(The reason I used a different message that omits “these stubs do nothing” was because I was worried people would just remove them without reading further and still hope to get the macOS 12 SDK. But I don’t think it’s a big deal, if we are always prepending that prefix anyway.)

Reply:
Awesome, removing the `prefix`/`suffix` shenanigans makes a lot of the messiness go away, yeah.